### PR TITLE
Add upgrade guide for forwarding rule validation

### PR DESCRIPTION
**Pull request by emilymye**
**Release Note Template for Downstream PRs (will be copied)**

### CODE REVIEW

File: third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown [hunk 1]
```diff
@@ -396,10 +398,10 @@ removed from Terraform to prevent conflicts.
 
 ```hcl
 resource "google_cloudiot_registry" "myregistry" {
-  name = "%s"
+  name = "my-registry"
 
   event_notification_config {
-    pubsub_topic_name = "${google_pubsub_topic.event-topic.id}"
+    pubsub_topic_name = $google_pubsub_topic.event-topic.id
```
Comment (danawillow) [inline third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown]:
```suggestion
    pubsub_topic_name = google_pubsub_topic.event-topic.id
```

File: third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown [hunk 2]
```diff
@@ -409,10 +411,61 @@ resource "google_cloudiot_registry" "myregistry" {
 
 ```hcl
 resource "google_cloudiot_registry" "myregistry" {
-  name = "%s"
+  name = "my-registry"
 
   event_notification_configs {
-    pubsub_topic_name = "${google_pubsub_topic.event-topic.id}"
+    pubsub_topic_name = $google_pubsub_topic.event-topic.id
```
Comment (danawillow) [inline third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown]:
```suggestion
    pubsub_topic_name = google_pubsub_topic.event-topic.id
```

File: third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown [hunk 3]
```diff
@@ -409,10 +411,61 @@ resource "google_cloudiot_registry" "myregistry" {
...
+
+Previously documentation suggested Terraform could use the same range of valid
+IP Address formats for `ip_address` as accepted by the API (e.g. named addresses
+or URLs to GCP Address resources). However, the server returns only literal IP
+addresses and thus caused diffs on re-apply (i.e. a permadiff). We amended
+documenation to say Terraform only accepts literal IP addresses.
+
+This is now strictly validated. While this shouldn't have a large breaking
+impact as users would have already run into permadiff issues on re-apply,
+there might be validation errors for existing configs. The solution is be to
```
Comment (danawillow) [inline third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown]:
```suggestion
there might be validation errors for existing configs. The solution is to
```

File: third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown [hunk 4]
```diff
@@ -409,10 +411,61 @@ resource "google_cloudiot_registry" "myregistry" {
...
+
+```hcl
+resource "google_compute_address" "my-addr" {
+  name = "my-addr"
+}
+
+resource "google_compute_forwarding_rule" "frule" {
+  name = "my-forwarding-rule"
+
+  address = $google_compute_address.my-addr.self_link
```
Comment (danawillow) [inline third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown]:
```suggestion
  address = google_compute_address.my-addr.self_link
```

File: third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown [hunk 5]
```diff
@@ -409,10 +411,61 @@ resource "google_cloudiot_registry" "myregistry" {
...
+
+```hcl
+resource "google_compute_address" "my-addr" {
+  name = "my-addr"
+}
+
+resource "google_compute_forwarding_rule" "frule" {
+  name = "my-forwarding-rule"
+
+  address = $google_compute_address.my-addr.address
```
Comment (danawillow) [inline third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown]:
```suggestion
  address = google_compute_address.my-addr.address
```

File: third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown (hunk 2 above)
Comment (emilymye) [reply third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown]: herp derp!

**Pull request merged**